boot/freeldr/freeldr/arch/i386/pcmem.c

```diff
@@ -286,6 +290,22 @@ PcMemGetBiosMemoryMap(PFREELDR_MEMORY_DESCRIPTOR MemoryMap, ULONG MaxMemoryMapSi
             break;
         }
 
+        /* Extra (input) safety. */
+        if (Regs.x.ecx < 20 || BiosMemoryMapSize < Regs.x.ecx)
+        {
+            ERR("BIOS returned an invalid entry length! (would-be-PcBiosMapCount = %lu, Entry length = (20 <=) %lu (<= %lu))\n\n",
+                PcBiosMapCount, Regs.x.ecx, BiosMemoryMapSize);
+            if (PcBiosMapCount > 0 || BiosMemoryMapSize < Regs.x.ecx)
+                ASSERT(20 <= Regs.x.ecx && Regs.x.ecx <= BiosMemoryMapSize);
```

Comment:
Since the assert condition is rather complicated and `ASSERT(FALSE);` is undesirable, using `ASSERTMSG("Descriptive text", 0);` might be appropriate.

Reply:
Thanks for the hint!
Done.